packages/react-core/src/components/Popover/PopoverHeader.tsx

```diff
@@ -31,23 +30,17 @@ export const PopoverHeader: React.FunctionComponent<PopoverHeaderProps> = ({
   id,
   alertSeverityScreenReaderText,
   ...props
-}: PopoverHeaderProps) => {
-  const HeadingLevel = titleHeadingLevel;
-
-  return icon || alertSeverityVariant ? (
-    <header className={css('pf-c-popover__header', className)} id={id} {...props}>
-      <HeadingLevel className={css(styles.popoverTitle)}>
-        {icon && <PopoverHeaderIcon>{icon}</PopoverHeaderIcon>}
+}: PopoverHeaderProps) => (
+  <header className={css('pf-c-popover__header', className)} id={id} {...props}>
+    <div className={css(styles.popoverTitle, icon && 'pf-m-icon')}>
```

Comment:
There's a difference in Core compared to React regarding the `id` here. In the latest Core update, `id` is placed on the `div.pf-c-popover__title` rather than `header.pf-c-popover__header`.

Reply:
For the `id` prop in PopoverHeader, it should be passed to `<div className={css(styles.popoverTitle, icon && 'pf-m-icon')}>` rather than PopoverHeaderText sub-component.

Core:

![Core Popover header](https://user-images.githubusercontent.com/70952936/222773208-0163130c-7b64-46d2-b1c7-f189cf9ee799.png)

Surge preview:

![Preview Popover header](https://user-images.githubusercontent.com/70952936/222773359-b4352ef2-52cb-4843-b419-88a7615a5995.png)